### PR TITLE
fix(core): `payment_method_type` not set in the payment attempt when making a recurring mandate payment

### PR DESCRIPTION
**Pull request by pixincreate**
## Type of Change

- [x] Bugfix

## Description

When doing a recurring mandate payment, the `payment_method` is set in the newly created payment attempt but the `payment_method_type` is not set. This PR aims to fix that.

**[#1378](https://github.com/juspay/hyperswitch/pull/1378) HAS TO BE MERGED BEFORE THIS PR IS MERGED AS THE CURRENT PR IS DEPENDENT ON THAT!**
**WHEN CREATING A MANDATE, MAKE SURE YOU THAT YOU GIVE `payment_method_type` IN THE REQUEST.**

## Motivation and Context

I fixed another bug! 

## How did you test it?

Manually.

### IMAGES
image
image
image

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

File: crates/router/src/core/payments/operations/payment_complete_authorize.rs [hunk 1]
```diff
@@ -110,10 +110,8 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Co
 
         payment_attempt.payment_method = payment_method.or(payment_attempt.payment_method);
         payment_attempt.browser_info = browser_info;
-        payment_attempt.payment_method_type = request
```
Comment (jagan-jaya) [inline crates/router/src/core/payments/operations/payment_complete_authorize.rs:113]: why are we ignoring the payment_method_type from request?

**Review by jagan-jaya — DISMISSED** [review]

File: crates/router/src/core/payments/operations/payment_confirm.rs [hunk 2]
```diff
@@ -129,10 +129,8 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Pa
 
         payment_attempt.payment_method = payment_method.or(payment_attempt.payment_method);
         payment_attempt.browser_info = browser_info;
-        payment_attempt.payment_method_type = request
```
[the remark this answers is not in the document]
Comment (pixincreate) [reply crates/router/src/core/payments/operations/payment_confirm.rs:132]: When we try to get the `payment_method_type` from the request, it always return a `None` for recurring mandate as we'll not be using the `payment_method_type` that we get from the `get_token_pm_type_mandate_details()`.

File: crates/router/src/core/payments/operations/payment_complete_authorize.rs (hunk 1 above)
Comment (pixincreate) [reply crates/router/src/core/payments/operations/payment_complete_authorize.rs:113]: When we try to get the `payment_method_type` from the request, it always return a `None` for recurring mandate as we'll not be using the `payment_method_type` that we get from the `get_token_pm_type_mandate_details()`.

**Review by vspecky — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]

**Review by vspecky — APPROVED** [review]

**Review by vspecky — CHANGES_REQUESTED** [review]
Actually, @jagan-jaya was correct in his previous two comments. We should also be taking the payment method type from the request. The precedence for taking the payment method type would be `mandate pmt OR request pmt OR original pmt from attempt`

**Review by vspecky — APPROVED** [review]
Ignore previous comment. Said comment is taken care of in the `helpers.rs` function

**Review by Narayanbhat166 — APPROVED** [review]

**Review by vspecky — APPROVED** [review]

**Pull request merged**